transport/cluster.go

```diff
@@ -100,9 +172,18 @@ func (c *Cluster) refreshTopology() error {
 	}
 	// If node is present in both maps we can reuse its connection pool.
 	old := c.Peers()
-	m := PeerMap{}
-	for _, v := range rows {
-		n, err := cqlRowIntoNode(v)
+	peers := make(PeerMap)
+	dcRacks := make(DCRacksMap)
+	seenRacks := make(map[string]bool) // Key is rack name appended to its dc name.
```

Comment:
Is this safe? It could confuse two racks as the same one, for example:
DC = "DC1", Rack = "11"
DC = "DC11", Rack = "1"
would both produce "DC111".
It's unlikely but maybe we should avoid it.

Reply:
I added "?" separator between DC and rack name, so now I don't think that this situation is likely enough to take it into consideration. Do you think it would still be better to just iterate through all nodes and check manually if a given combination of DC and rack has already been encountered?